support/src/main/java/org/jenkinsci/plugins/workflow/support/pickles/FilePathPickle.java

```diff
@@ -86,6 +86,18 @@ protected FilePath tryResolve() {
         // TODO better to use a synchronized accessor
         @Restricted(NoExternalUse.class)
         public static final Map<VirtualChannel,String> channelNames = new WeakHashMap<VirtualChannel,String>();
+
+        static {
+            // Add channels that may already be known to Jenkins. This can be the case
+            // just after the Workflow plugins are installed (see JENKINS-25958).
+            Jenkins instance = Jenkins.getInstance();
+            if (instance != null) {
+                for (Computer c : instance.getComputers()) {
+                    channelNames.put(c.getChannel(), c.getName());
```

Comment:
> the existing code was not checking for it

Because the existing code was in a listener callback which is called as soon as the channel is constructed and set. This code is not.